hazelcast/src/main/java/com/hazelcast/config/TSDiskTierConfig.java

```diff
@@ -0,0 +1,206 @@
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * Disk tier configuration of Tiered-Store.
+ *
+ * @since 5.1
+ */
+public class TSDiskTierConfig implements IdentifiedDataSerializable {
+
+    /**
+     * Default base directory for the tiered-store.
+     */
+    public static final String DEFAULT_TSTORE_BASE_DIR = "tstore";
+
+    /**
+     * Default block/sector size in bytes.
+     */
+    public static final int DEFAULT_BLOCK_SIZE_IN_BYTES = 4096;
+
+    private boolean enabled;
+    private File baseDir = new File(DEFAULT_TSTORE_BASE_DIR).getAbsoluteFile();
+    private int blockSize = DEFAULT_BLOCK_SIZE_IN_BYTES;
+    private long capacity;
+
+    public TSDiskTierConfig() {
+
+    }
+
+    public TSDiskTierConfig(TSDiskTierConfig tsDiskTierConfig) {
+        enabled = tsDiskTierConfig.isEnabled();
+        baseDir = new File(tsDiskTierConfig.getBaseDir().getAbsolutePath());
+        blockSize = tsDiskTierConfig.getBlockSize();
+        capacity = tsDiskTierConfig.getCapacity();
+    }
+
+    /**
+     * Returns whether disk tier is enabled on the related tiered-store.
+     *
+     * @return true if disk tier is enabled, false otherwise.
+     */
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    /**
+     * Sets whether disk tier is enabled on the related tiered-store.
+     *
+     * @param enabled enabled parameter.
+     * @return this TSDiskTierConfig
+     */
+    public TSDiskTierConfig setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        return this;
+    }
+
+    /**
+     * Base directory for this disk tier.
+     * Can be an absolute or relative path to the node startup directory.
+     */
+    public File getBaseDir() {
+        return baseDir;
+    }
+
+    /**
+     * Sets the base directory for this disk tier.
+     *
+     * @param baseDir base directory.
+     * @return this TSDiskTierConfig
+     */
+    public TSDiskTierConfig setBaseDir(File baseDir) {
+        this.baseDir = baseDir;
+        return this;
+    }
+
+    /**
+     * Returns disk block/sector size in bytes.
+     *
+     * @return block size
+     */
+    public int getBlockSize() {
+        return blockSize;
+    }
+
+    /**
+     * Sets the disk block/sector size in bytes.
+     * @param blockSize block size.
+     * @return this TSDiskTierConfig
+     */
+    public TSDiskTierConfig setBlockSize(int blockSize) {
+        this.blockSize = blockSize;
+        return this;
+    }
+
+    /**
+     * Returns the capacity of this disk tier.
+     *
+     * @return disk tier capacity.
+     */
+    public long getCapacity() {
+        return capacity;
+    }
+
+    /**
+     * Sets the capacity of this disk tier.
+     *
+     * @param capacity capacity.
+     * @return this TSDiskTierConfig
+     */
+    public TSDiskTierConfig setCapacity(long capacity) {
```

Comment:
It can be  helpful if we put a short answer in javaDoc  of `setCapacity` for this potential issue:
what will happen if i set more capacity then i have? Fail-fast immediately or fail at runtime later?

Reply:
Thanks, good question. Fail-fast might not be plausible, since disk free space check is not much helpful. Failing at runtime will probably be the way to go. @petrpleshachkov what will we do when we run out of disk space for tiered-store: throw exception and terminate HZ instance or remove oldest entries from the disk and continue execution (maybe it can be sth configurable by the user) ?

Reply:
ok, we will crash at runtime after running out of disk space and since we will no longer have `capacity` for the disk-tier (for now), I am resolving this comment.